test/configuration/ssh_test.rb

```diff
@@ -37,4 +37,24 @@ class ConfigurationSshTest < ActiveSupport::TestCase
     config = Kamal::Configuration.new(@deploy.tap { |c| c.merge!(ssh: { "proxy" => "app@1.2.3.4" }) })
     assert_equal "app@1.2.3.4", config.ssh.options[:proxy].jump_proxies
   end
+
+  test "ssh key_data with plain value array" do
+    config = Kamal::Configuration.new(@deploy.tap { |c| c.merge!(ssh: { "key_data" => ["-----BEGIN OPENSSH PRIVATE KEY-----"] }) })
```

Comment:
if the suggestion is accepted, the test become strictier 

you will need to generate a real ssh key
```ruby
key = OpenSSL::PKey::RSA.new(2048)
  key_pem = key.to_pem

config = Kamal::Configuration.new(@deploy.tap { |c| c.merge!(ssh: { "key_data" => [key_pem] }) })
resolved_key = config.ssh.options[:key_data].first

assert_nothing_raised { OpenSSL::PKey::RSA.new(resolved_key) }